spec/models/enterprise_relationship_spec.rb

```diff
@@ -255,5 +255,87 @@
         end
       end
     end
+    describe "updating order cycles" do
+      let(:hub) { create(:distributor_enterprise) }
+      let(:producer) { create(:supplier_enterprise) }
+      let(:order_cycle) { create(:simple_order_cycle) }
+      let(:some_other_producer) { create(:supplier_enterprise) }
+
+      context "when order_cycle permission is present" do
+        let!(:er) {
+          create(:enterprise_relationship, child: hub, parent: producer,
+                                           permissions_list: [:add_to_order_cycles, :create_variant_overrides] )
+        }
+        let!(:incoming_external_exchange) {
+          order_cycle.exchanges.create! sender: producer, receiver: hub, incoming: true
+        }
+        let!(:other_external_exchange) {
+          order_cycle.exchanges.create! sender: some_other_producer, receiver: hub, incoming: true
+        }
+        let!(:incoming_internal_exchange) {
+          order_cycle.exchanges.create! sender: hub, receiver: hub, incoming: true
+        }
+        let!(:outgoing_internal_exchange) {
+          order_cycle.exchanges.create! sender: hub, receiver: hub, incoming: false
+        }
+        let!(:variant) { create(:variant) }
+        let!(:some_other_variant) { create(:variant) }
+        let!(:incoming_external_variant) {
+          incoming_external_exchange.exchange_variants.create! exchange: incoming_external_exchange, variant: variant
+        }
+        let!(:incoming_internal_only_variant) {
+          incoming_internal_exchange.exchange_variants.create! exchange: incoming_internal_exchange, variant: some_other_variant
+        }
+        let!(:outgoing_internal_variant) {
+          outgoing_internal_exchange.exchange_variants.create! exchange: outgoing_internal_exchange, variant: variant
+        }
+        let!(:outgoing_internal_only_variant) {
+          outgoing_internal_exchange.exchange_variants.create! exchange: outgoing_internal_exchange, variant: some_other_variant
+        }
+
+        context "removing exchanges and exchange variants" do
+          context "when the enterprise relationship is destroyed" do
+            before { er.destroy }
+            it "should destroy all exchanges and exchange variants related to exchange relationship" do
+              expect(Exchange.exists?(incoming_external_exchange.id)).to be false
+              expect(Exchange.exists?(other_external_exchange.id)).to be true
+              expect(ExchangeVariant.exists?(incoming_external_variant.id)).to be false
+              expect(ExchangeVariant.exists?(outgoing_internal_variant.id)).to be false
+              expect(ExchangeVariant.exists?(incoming_internal_only_variant.id)).to be true
+              expect(ExchangeVariant.exists?(outgoing_internal_only_variant.id)).to be true
+            end
+          end
+        end
+
+        context "and is then removed" do
+          before { er.permissions_list = [:create_variant_overrides]; er.save! }
```

Comment:
I think this would be clearer:

```suggestion

          before { er.permissions_list -= [:add_to_order_cycles]; er.save! }

```

Reply:
Can I write a function that takes a custom operator? I tried to write something like [what is written for the assignment operator](https://github.com/openfoodfoundation/openfoodnetwork/blob/ad9e82e97364de2d4f088834eb2e954b5d072b7e/app/models/enterprise_relationship.rb#L74), but no luck. What do you think of the following?
```
er.revoke_permission!(:add_to_order_cycle)
```
and define the function in `app/models/enterprise_relationship.rb`

Reply:
Perfect!